cli/src/commands/explore/mod.rs

```diff
@@ -1,159 +1,223 @@
 /// ```
-/// ## Storage
+/// ### Storage
 ///
 /// Show the storage entries in a pallet
 /// ```text
-/// subxt explore Alliance storage
+/// subxt explore pallet Alliance storage
 /// ```
 /// Show the types and value of a specific storage entry
 /// ```text
-/// subxt explore Alliance storage Announcements [KEY_SCALE_VALUE]
+/// subxt explore pallet Alliance storage Announcements [KEY_SCALE_VALUE]
 /// ```
-///
-#[derive(Debug, ClapParser)]
+/// ### Events
+/// ```text
+/// subxt explore pallet Balances events
+/// ```
+/// Show the type of a specific event
+/// ```text
+/// subxt explore pallet Balances events frozen
+/// ```
+/// ## Runtime APIs
+/// Show the input and output types of a runtime api:
+/// ```text
+/// subxt explore api core version
+/// ```
+/// Execute a runtime API call with the `--execute` (`-e`) flag, to see the return value:
+/// ```text
+/// subxt explore api core version --execute
+/// ```
+#[derive(Debug, Parser)]
 pub struct Opts {
     #[command(flatten)]
     file_or_url: FileOrUrl,
-    pallet: Option<String>,
     #[command(subcommand)]
-    pallet_subcommand: Option<PalletSubcommand>,
+    subcommand: Option<PalletOrRuntimeApi>,
     /// Allow insecure URLs e.g. URLs starting with ws:// or http:// without SSL encryption
     #[clap(long, short)]
     allow_insecure: bool,
 }
 
-#[derive(Debug, Clone, Subcommand)]
-pub enum PalletSubcommand {
-    Calls(CallsSubcommand),
-    Constants(ConstantsSubcommand),
-    Storage(StorageSubcommand),
+#[derive(Debug, Subcommand)]
+pub enum PalletOrRuntimeApi {
+    Pallet(PalletOpts),
+    Api(RuntimeApiOpts),
+}
+
+#[derive(Debug, Parser)]
+pub struct PalletOpts {
+    pub name: Option<String>,
+    #[command(subcommand)]
+    pub subcommand: Option<PalletSubcommand>,
+}
+
+#[derive(Debug, Parser)]
+pub struct RuntimeApiOpts {
+    pub name: Option<String>,
+    #[clap(required = false)]
+    pub method: Option<String>,
+    #[clap(long, short, action)]
+    pub execute: bool,
+    #[clap(required = false)]
+    trailing_args: Vec<String>,
 }
 
 pub async fn run(opts: Opts, output: &mut impl std::io::Write) -> color_eyre::Result<()> {
     validate_url_security(opts.file_or_url.url.as_ref(), opts.allow_insecure)?;
 
     // get the metadata
-    let bytes = opts.file_or_url.fetch().await?;
+    let file_or_url = opts.file_or_url;
+    let bytes = file_or_url.fetch().await?;
     let metadata = Metadata::decode(&mut &bytes[..])?;
 
-    // if no pallet specified, show user the pallets to choose from:
-    let Some(pallet_name) = opts.pallet else {
-        let available_pallets = print_available_pallets(&metadata);
-        writeln!(output, "Usage:",)?;
-        writeln!(output, "    subxt explore <PALLET>",)?;
-        writeln!(output, "        explore a specific pallet",)?;
-        writeln!(output, "\n{available_pallets}",)?;
-        return Ok(());
-    };
+    #[allow(non_snake_case)]
+    let PALLET: String = "<PALLET>".blue().to_string();
```

Comment:
I don't get why you use capital letters on PALLET and RUNTIME_API, can you explain?

Reply:
I use these strings like constants, but the `blue()` function is not const. So I use these uppercase variable names to not confuse it with something else.

I wish Rust would automatically have functions const by default, much like a type is Send as long as it does not have any non Send fields. Maybe they can put all of const evaluation behind a `Const` auto-trait in the future that all functions implement except the ones doing non-const stuff.

Reply:
As I said elsewhere, I'd instead call them eg `blue_input_placeholder` and then no need for the opt-out allow. I don't think it's worth "breaking the rules" for this case :)